### PR TITLE
[Client] Filter detected rectangles

### LINKED ISSUE #7: [Client] [CV] Search for lightweight ways to filter rectangles that are certainly not pictures. Should be based only on computer vision techniques and heuristics.

### LINKED ISSUE #21: [Client] [CV] Using the Swift computer vision API, implement the core client functionality which detects rectangles.

### PR DESCRIPTION
**Pull request by AndreiNet**

### CODE REVIEW

File: app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift [hunk 1]
```diff
@@ -168,19 +165,69 @@ class RectangleDetectionViewController: CameraViewController {
...
     
     private func filterRectangles(rectangles: [Detection]) -> [Detection] {
-        var rectangles = filterRectanglesByBounds(rectangles: rectangles)
-        rectangles = filterRectanglesByInclusion(rectangles: rectangles)
+        let rectangles = filterRectanglesByBounds(rectangles: rectangles)
+//        rectangles = filterRectanglesByInclusion(rectangles: rectangles)
         return rectangles
     }
     
     private func filterRectanglesByBounds(rectangles: [Detection]) -> [Detection] {
```
Comment (darius98) [inline app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift]: Either rename the method to not contain Rectangles in the name, or make the argument nameless.
```swift
filterByBounds(rectangles: rectangles)
```
or
```swift
filterRectanglesByBounds(rectangles)

func filterRectanglesByBounds(_ rectangles: [Detection]) -> [Detection]
```

File: app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift [hunk 2]
```diff
@@ -168,19 +165,69 @@ class RectangleDetectionViewController: CameraViewController {
...
+                let closeVertical =
+                    (abs(p.x - left) < bounds.width * ratio || abs(p.x - right) < bounds.width * ratio)
+                        && top - bounds.height * ratio < p.y && p.y < bottom + bounds.height * ratio
+                return closeHorizontal || closeVertical
+            }.count == rectangle.points.count
+        }
     }
     
+    // This is not used at the moment, but I will leave it here just in case
     private func filterRectanglesByInclusion(rectangles: [Detection]) -> [Detection] {
```
Comment (darius98) [inline app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift]: Same as above. Too many rectangles.

File: app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift [hunk 3]
```diff
@@ -168,19 +165,69 @@ class RectangleDetectionViewController: CameraViewController {
...
+            for otherDetection in rectangles {
+                let otherRectangle = otherDetection.relocated
+                if rectangle == otherRectangle {
+                    continue
+                }
+                var rectangleOutside = false
+                for point1 in rectangle.points {
+                    var pointOutside = false
+                    for i in 0..<4 {
+                        print(crossProduct(otherRectangle.points[i], otherRectangle.points[(i + 1) % 4], point1))
```
Comment (darius98) [inline app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift]: Do you need this print to stay here in production?

**Review by darius98 — APPROVED** [review]
LGTM

File: app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift (hunk 3 above)
Comment (AndreiNet) [reply app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift]: No, I removed it.

File: app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift (hunk 1 above)
Comment (AndreiNet) [reply app/Pain/Pain/Interface/Camera/RectangleDetectionViewController.swift]: I renamed some things

**Pull request merged**